File: src/Application/Interfaces/Services/SessionAssistees/SpeAggregate.cs
namespace Application.Interfaces.Services.SessionAssistees;

public record SpeAggregate(
    IDictionary<byte, int> SessionActuelle,
    IDictionary<byte, double> MoyenneSessionsPrecedentes,
    IEnumerable<IDictionary<byte, int>> TroisDernieresSessions);

File: src/Web/vue-app/src/types/responses/speAggregate.ts
export type SPE = number;
export type NbEtudiants = number;

export type SpePourSession = Record<SPE, NbEtudiants>;

export class SpeAggregate {
    sessionActuelle = {} as SpePourSession;
    moyenneSessionsPrecedentes = {} as SpePourSession;
    troisDernieresSessions = [] as SpePourSession[];
}